mephisto/abstractions/providers/prolific/prolific_worker.py

```diff
@@ -0,0 +1,315 @@
+        logger.debug(f'{self.log_prefix}Blocking worker {self.worker_name}')
+
+        if not unit and not requester:
+            # TODO(WISH) soft block from all requesters? Maybe have the main requester soft block?
+            return (
+                False,
+                'Blocking without a unit or requester not yet supported for ProlificWorkers',
+            )
+        elif unit and not requester:
+            task_run = unit.get_assignment().get_task_run()
+            requester: 'ProlificRequester' = cast('ProlificRequester', task_run.get_requester())
+        else:
+            task_run = self._get_first_task_run(requester)
+
+        logger.debug(f'{self.log_prefix}Task Run: {task_run}')
+
+        task_run_args = task_run.args
+        requester: 'ProlificRequester' = cast('ProlificRequester', requester)
+        client = self._get_client(requester.requester_name)
+        prolific_utils.block_worker(client, task_run_args, self.worker_name, reason)
+        self.datastore.set_worker_blocked(self.worker_name, is_blocked=True)
+
+        # Find all previously granted qualifications for this worker,
+        # and remove the worker from all related Prolific Participant Groups
+        db_granted_qualifications = self.db.find_granted_qualifications(worker_id=self.db_id)
+        if db_granted_qualifications:
+            db_qualification_ids = [q.qualification_id for q in db_granted_qualifications]
+            prolific_qualifications = self.datastore.find_qualifications_for_running_studies(
+                db_qualification_ids,
+            )
+            prolific_participant_group_ids = [
+                p['prolific_participant_group_id'] for p in prolific_qualifications
+            ]
+            for prolific_participant_group_id in prolific_participant_group_ids:
+                prolific_utils.remove_worker_qualification(
+                    client, self.worker_name, prolific_participant_group_id,
+                )
+
+        logger.debug(f'{self.log_prefix}Worker {self.worker_name} blocked')
+
+        return True, ''
+
+    def unblock_worker(self, reason: str, requester: 'Requester') -> Tuple[bool, str]:
+        """Unblock a blocked worker for the specified reason. Return success of unblock"""
+        logger.debug(f'{self.log_prefix}Unlocking worker {self.worker_name}')
+
+        task_run = self._get_first_task_run(requester)
+
+        logger.debug(f'{self.log_prefix}Task Run: {task_run}')
+
+        task_run_args = task_run.args
+        requester = cast('ProlificRequester', requester)
+        client = self._get_client(requester.requester_name)
+        prolific_utils.unblock_worker(client, task_run_args, self.worker_name, reason)
+        self.datastore.set_worker_blocked(self.worker_name, is_blocked=False)
+
+        # Include unblocked worker into all currently running studies, if qualified
+        self._grant_crowd_qualification(client, requester, task_run_args.provider)
```

Comment:
it's not inherently clear what this `_grant_crowd_qualification` call is doing, given no qualification name appears provided.